flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/DefaultScheduler.java

```diff
@@ -235,6 +236,9 @@ public void allocateSlotsAndDeploy(final Collection<ExecutionVertexDeploymentOpt
 		final Map<ExecutionVertexID, ExecutionVertexDeploymentOption> deploymentOptionsByVertex =
 			groupDeploymentOptionsByVertexId(executionVertexDeploymentOptions);
 
+		checkArgument(executionVertexDeploymentOptions.size() == deploymentOptionsByVertex.size(),
```

Comment:
I think this should be part of `groupDeploymentOptionsByVertexId()`. However, I believe it is not needed:

```
public static void main(String[] args) {
		Arrays
			.asList("bla", "bla")
			.stream()
			.collect(Collectors.toMap(
				Function.identity(),
				Function.identity()));
	}
```

```
Exception in thread "main" java.lang.IllegalStateException: Duplicate key bla
	at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
	at java.util.HashMap.merge(HashMap.java:1254)
	at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566)
	at org.apache.flink.runtime.scheduler.DefaultScheduler.main(DefaultScheduler.java:285)
```

If you want, you can add a unit test.

Reply:
You are right. We can drop this change.

Here's the java doc of `Collectors.toMap()`
> <p>If the mapped keys contains duplicates (according to
>      * {@link Object#equals(Object)}), an {@code IllegalStateException} is
>      * thrown when the collection operation is performed.  If the mapped keys
>      * may have duplicates, use {@link #toMap(Function, Function, BinaryOperator)}
>      * instead.

Reply:
Done.